src/acvp_kts_ifc.c

```diff
@@ -81,23 +72,29 @@ static ACVP_RESULT acvp_kts_ifc_init_tc(ACVP_CTX *ctx,
                                             const char *d,
                                             const char *n,
                                             const char *e,
+                                            const char *dmp1,
+                                            const char *dmq1,
+                                            const char *iqmp,
                                             int modulo, 
                                             int llen, 
                                             ACVP_KTS_IFC_TEST_TYPE test_type) {
     ACVP_RESULT rv;
 
-    stc->llen = llen/8;
+    stc->llen = llen / 8;
     stc->modulo = modulo;
     stc->test_type = test_type;
     stc->md = hash_alg;
     stc->kts_role = role;
     stc->key_gen = key_gen;
 
+    stc->ct = calloc(1, ACVP_KTS_IFC_BYTE_MAX);
+    if (!stc->ct) { return ACVP_MALLOC_FAIL; }
+
+    stc->pt = calloc(1, ACVP_KTS_IFC_BYTE_MAX);
+    if (!stc->pt) { return ACVP_MALLOC_FAIL; }
```

Comment:
if you return here, do you need to free stc->ct ?

Reply:
the test case gets passed back to the caller which eventually frees any non-null pointers in its own function (acvp_kts_ifc_release_tc)